src/components/pages/lessons/overlay/go-pro-cta-overlay.tsx

```diff
@@ -2,30 +2,170 @@ import React, {FunctionComponent} from 'react'
+import {PlanPrice} from 'components/pricing/select-plan-new/index'
+import SmallParityCouponMessage from './small-parity-coupon-message'
+import {Coupon} from 'machines/commerce-machine'
+import get from 'lodash/get'
+import isEmpty from 'lodash/isEmpty'
+import first from 'lodash/first'
+import {useViewer} from 'context/viewer-context'
+import emailIsValid from 'utils/email-is-valid'
+import axios from 'axios'
+import stripeCheckoutRedirect from 'api/stripe/stripe-checkout-redirect'
+import toast from 'react-hot-toast'
+import {StripeAccount} from 'types'
 
 type JoinCTAProps = {
   lesson: LessonResource
 }
 
 const GoProCtaOverlay: FunctionComponent<JoinCTAProps> = ({lesson}) => {
+  const {viewer, authToken} = useViewer()
+  const {state, send, priceId, quantity, availableCoupons, currentPlan} =
+    useCommerceMachine({initialPlan: 'monthlyPrice'})
+
+  const [emailForCheckout, setEmailForCheckout] = React.useState<
+    string | undefined
+  >(viewer?.email)
+
+  React.useEffect(() => {
+    if (!isEmpty(viewer?.email)) {
+      setEmailForCheckout(viewer.email)
+    }
+  }, [viewer?.email])
+
+  const pricesLoading = !state.matches('pricesLoaded')
+  const pppCouponIsApplied =
+    state.matches('pricesLoaded.withPPPCoupon') ||
+    (pricesLoading && state.context?.couponToApply?.couponType === 'ppp')
+
+  // machine-derived data
+  const parityCoupon = availableCoupons?.['ppp']
+
+  const countryCode = get(parityCoupon, 'coupon_region_restricted_to')
+  const countryName = get(parityCoupon, 'coupon_region_restricted_to_name')
+
+  const pppCouponAvailable =
+    !isEmpty(countryName) && !isEmpty(countryCode) && !isEmpty(parityCoupon)
+  const pppCouponEligible = quantity === 1
+
+  // handlers
+  const onApplyParityCoupon = () => {
+    send('APPLY_PPP_COUPON')
+  }
+
+  const onDismissParityCoupon = () => {
+    send('REMOVE_PPP_COUPON')
+  }
+
+  const onClickCheckout = async () => {
+    if (!priceId) return
+    if (!emailForCheckout) return
+
+    const account = first<StripeAccount>(get(viewer, 'accounts'))
+    await track('checkout: selected plan', {
+      priceId: priceId,
+    })
+
+    const emailRequiresAProCheck =
+      !isEmpty(emailForCheckout) && emailForCheckout !== viewer?.email
+
+    if (emailRequiresAProCheck) {
+      const {hasProAccess} = await axios
+        .post(`/api/users/check-pro-status`, {
+          email: emailForCheckout,
+        })
+        .then(({data}) => data)
+
+      if (hasProAccess) {
+        const message = `You've already got a pro account at ${emailForCheckout}. Please sign in.`
+
+        toast.error(message, {
+          duration: 6000,
+          icon: '❌',
+        })
+
+        track('cta overlay checkout: existing pro account found', {
+          email: emailForCheckout,
+        })
+
+        // email is already associated with a pro account, return early instead
+        // of sending the user to a Stripe Checkout Session.
+        return
+      }
+    }
```

Comment:
Similar to the _Step 2: Email_ form that we inject between the Pricing Page and Stripe Checkout, we do a Pro Access Check on the user-provided email.